### PR TITLE
[TS] Guidelines for TS core libraries, plus structure updates, cleanups

### PR DESCRIPTION
**Pull request by bterlson**

### CODE REVIEW

File: docs/design/typescript/AzureCore.mdk [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
+## Azure Core Libraries for TypeScript {#ts-azure-core}
```
Comment (bterlson) [inline docs/design/typescript/AzureCore.mdk]: This file is the meat you probably want to look at.

File: docs/design/typescript/DesignGuidelinesNew.mdk [hunk 2]
```diff
@@ -0,0 +1,87 @@
+Madoko documentation: http://madoko.org/reference.html
+
+Title       : Azure SDK Design Guidelines for TypeScript
+Title Note  : Version 1.0.0-preview2
```
Comment (adrianhall) [inline docs/design/typescript/DesignGuidelinesNew.mdk]: File should really be DesignGuidelines.mdk and we need to lose the preview2 from the PR as, once approved, they become v1.0.0

Comment (bterlson) [reply docs/design/typescript/DesignGuidelinesNew.mdk]: I will do so once I'm done moving everything over. This is a point-in-time thing :)

File: docs/design/typescript/AzureCore.mdk [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+~ TableFigure {#ts-core-components-list, caption: "Core components of the Azure SDK for TS & JS"}
+
+| Package | Purpose |
+|---------|---------|
+| core-http | Implements the HTTP pipeline.<br>Previously called ms-rest-js. |
+| core-amqp | Implements the AMQP pipeline and shared functionality.<br> Previously called amqp-common. |
+| core-lro | Abstraction and types for implementing long-running operations. |
+| core-paging | Types for implementing paginated APIs. |
+| abort-controller | Implementation of AbortSignal/AbortController for cancelling async operations. |
+| sdk-configuration | Common configuration singleton for the Azure SDK (not yet implemented). |
```
[the remark this answers is not in the document]
Comment (bterlson) [reply docs/design/typescript/AzureCore.mdk]: See the following line for rationale :)

**Comment by bterlson** [comment]
This is superceded by #424.

**Pull request closed without merging**